src/common/dump.cpp

```diff
@@ -44,6 +44,44 @@ void dump_to_xflash(void) {
     w25x_fetch_error();
 }
 
+void dump_err_to_xflash(const char *error, const char *title) {
+    dumpmessage_t dump_message;
+    w25x_sector_erase(w25x_error_start_adress);
+    dump_message.flag = 0;
+    strlcpy(dump_message.msg_title, title, sizeof(dump_message.msg_title));
+    strlcpy(dump_message.msg_content, error, sizeof(dump_message.msg_content));
+    w25x_program(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    w25x_fetch_error();
+}
+
+int dump_in_xflash_get_message(char *msg_dst, char *title_dst) {
+    dumpmessage_t dump_message;
+    w25x_rd_data(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    if (w25x_fetch_error())
+        return 0;
+    strlcpy(title_dst, dump_message.msg_title, sizeof(dump_message.msg_title));
+    strlcpy(msg_dst, dump_message.msg_content, sizeof(dump_message.msg_content));
+    return 1;
+}
+
+void dump_in_xflash_clear_message_flag(void) {
+    uint8_t dumpmessage_flag;
+    w25x_rd_data(w25x_error_start_adress, &dumpmessage_flag, 1);
+    if (!w25x_fetch_error() && dumpmessage_flag == 0) {
+        dumpmessage_flag = 0xFF;
+        w25x_program(w25x_error_start_adress, &dumpmessage_flag, 1);
```

Comment:
You can not write to the same flash location twice without erasing whole sector. You already written there 0 in dump_err_to_xflash().

Reply:
I would suggest change
```
typedef struct _dumpmessage_t {
    unsigned short flag; // flags if message is valid (== 0x00) or not valid (!= 0x00)
    char msg_title[DUMP_MSG_TITLE_MAX_LEN];
    char msg_content[DUMP_MSG_MAX_LEN];
} dumpmessage_t;
```
to
```
typedef struct _dumpmessage_t {
    uint8_t invalid; // flags if message is valid (== 0x00) or invalid (!= 0x00)
    uint8_t not_displayed; // flags if message was not displayed (== 0xff) or displayed (== 0x00)
    char msg_title[DUMP_MSG_TITLE_MAX_LEN];
    char msg_content[DUMP_MSG_MAX_LEN];
} dumpmessage_t;
```
And then in `dump_err_to_xflash()` just write `.invalid` to 0 and don't touch `.not_displayed`. That way `.not_displayed` would be kept erased ==0xff meaning not displayed. You can switch it later in dump_in_xflash_clear_message_flag to 0, so `.not_displayed` would be turned to displayed.

Reply:
Or instead of flag `invalid` you can call it `version` and you can start with `version == 0` (with all the logic being same as if it would be `invalid`). That way if someone decides to alter `dumpmessage_t` he can increment `version`, so if there would be something stored with older version after firmware updatate, instead of displaying untangible nonsense later firmware would ignore data stored with older version format.

Reply:
`7c21391`